### ISSUE TITLE
Promises Coming Back As Gibberish?

### ISSUE THREAD

**Issue by BobbieBarker** [bvaughn/jasmine-es6-promise-matchers#3]
When I try to unit test my promises and console log the promise object it looks like this:

 b{_handler: v{consumers: undefined, receiver: undefined, handler: z{id: ..., value: ..., handled: ..., reported: ...}, resolved: true}}

None of the methods from this lib can do anything with them.
When I try the more old school way of running promises as detailed in your readme:

var resolvedValue;

promise.then(
  function(value) {
    resolvedValue = value;
  });

jasmine.clock().tick(1);

expect(resolvedValue).toBe('something');

resolvedvalue is always undefined.

--When I use my app it works just fine, all the promises are behaving as expected. 

What am I doing wrong?

**Comment by bvaughn** [comment]
Hm. Can you point me at a Plunker/JSBin/Fiddle?

**Comment by BobbieBarker** [comment]
I don't know how to run tests in a plunk. Do you know of any boiler plate plunk that is set up with karma etc?

**Comment by bvaughn** [comment]
Here's some examples you could potentially fork:
* JSFiddle: http://jsfiddle.net/eitanp461/E64Se/
* Plunker: http://embed.plnkr.co/UfNWsJ/preview

Or you could maybe point me at a branch in Github (or wherever) and I could just check it out myself?

**Comment by BobbieBarker** [comment]
First I'd like to thank you for taking the time to respond. Second, setting up a plunkr to run karma and unit test es6 code is a super hard challenge. One at which I admittedly failed. 
My best effort can be witnessed here:
http://plnkr.co/edit/JEnEpr?p=preview

The best course of action might be to pull the query and query.spec files into a local enviroment set up for es6 testing.

**Comment by bvaughn** [comment]
No problem. :)

Let me take a look.

**Comment by bvaughn** [comment]
Hmm, you've got some ES6 in there that's not going to run directly in a browser (Chrome doesn't support the `class` keyword and Firefox doesn't support `let`...)

Looks like there's also a typo in the last test (*should return a query Obj*). Should be `test.compose`, not `query.compose`.

That being said, I'm not sure what's going on... :/ If you'd want to toss up a zip file of a project or something somewhere, I'll check it out.

**Comment by BobbieBarker** [comment]
I am running the test through traceur/jspm in my karma .conf also all my
tests before that work. Sorry for the typo that's not in my project code.

**Comment by BobbieBarker** [comment]
Hey I forgot to ask if you where able to replicate my results or if it worked for you.

**Comment by bvaughn** [comment]
The tests did not work for me, but I didn't spend a lot of time investigating. (I just moved this weekend and I've been pretty busy between work and unpacking.) From what I saw, setting breakpoints everywhere, it looked like the (native) Promises were resolving/rejecting - but not until *after* the test had failed. I'm not sure why this is, or why it's different from projects I've written that used the Jasmine matchers. I'll try to investigate further when I get more time but it will probably be a few days.

**Comment by bvaughn** [comment]
I've released an updated version of this library (2.0) that should hopefully address your concerns. Note that a small signature change has been made to the matchers (as shown in the README/examples).

Here's an updated version of your tests that should work if run against the new release:
```js
describe('someClass Class', function(){
  var test = {};

  beforeEach(() => {
    JasminePromiseMatchers.install();
    test = new someClass();
  })
 
  afterEach(JasminePromiseMatchers.uninstall);
 
  it('should exist', () => {
    expect(test).toBeDefined();
  })
 
  it('should possess a compose function', () => {
    expect(test.compose).toBeDefined();
    expect(test.compose).toEqual(jasmine.any(Function));
  })
 
  it('should reject an empty arguement', (done) => {
    let promise = test.compose({});
    expect(promise).toBeRejected(done);
  })
 
  it('should return a query Obj', (done) => {
    let idealStuff = {name: 'superQ'};
    let promise = test.compose(idealStuff);
    let partialMatch = jasmine.objectContaining({name: 'superQ'});
    expect(promise).toBeResolvedWith(partialMatch, done);
  })
});
```


Thanks for bringing up this issue. I apologies that it took me a few days to get around to it. I've been busy unpacking. :)
